app/lib/analyzer/models.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+import 'package:pub_semver/pub_semver.dart';
+
+/// These status codes mark the status of the analysis, not the result/report.
+class AnalysisStatus {
+  /// Analysis was aborted without a report.
+  /// Probably an issue with pana or an unhandled edge case.
+  static const aborted = 'aborted';
+
+  /// Analysis was completed but there are missing parts.
+  /// One or more tools failed to produce the expected output.
+  static const failure = 'failure';
+
+  /// Analysis was completed without issues.
+  static const success = 'success';
+}
+
+/// Fields that contain the properties of the latest analysis.
+///
+/// For [PackageAnalysis] the rollup is collected only for the latest stable
+/// version.
+class AnalysisRollup {
+  @db.DateTimeProperty()
+  DateTime analysisTimestamp;
+
+  @db.ModelKeyProperty()
+  db.Key analysisKey;
+
+  db.Key get analyzedVersionKey => analysisKey.parent;
+  Version get semanticAnalyzedVersion =>
+      new Version.parse(analyzedVersionKey.id);
+
+  @db.StringProperty()
+  String analysisVersion;
+
+  @db.StringProperty()
+  String analysisStatus;
+}
+
+/// Analyzed package report.
+@db.Kind(name: 'PackageAnalysis', idType: db.IdType.String)
+class PackageAnalysis extends db.ExpandoModel with AnalysisRollup {
+  @db.StringProperty(required: true)
+  String package; // Same as id
+
+  /// See also [AnalysisRollup] fields.
+}
+
+/// Analyzed version report.
+@db.Kind(name: 'PackageVersionAnalysis', idType: db.IdType.String)
+class PackageVersionAnalysis extends db.ExpandoModel with AnalysisRollup {
+  @db.ModelKeyProperty(required: true, propertyName: 'package')
+  db.Key packageKey;
+
+  String get package => packageKey.id;
+
+  @db.StringProperty(required: true)
+  String version; // Same as id
+
+  /// See also [AnalysisRollup] fields.
```

Comment:
If you look at the class, if is clear that it inherits fields from another class. However, if you just glance at the list of fields, it may not be obvious. I thought a reminder will be nice there.